public_html/wp-content/themes/pattern-directory/src/components/report-pattern-modal/index.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useReducer, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, RadioControl, Spinner, TextareaControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { store as patternStore } from '../../store';
+
+const reducer = ( state, action ) => {
+	switch ( action.status ) {
+		case 'submitting':
+			return { isSubmitting: true };
+		case 'submitted':
+			return { isSubmitted: true };
+		case 'error':
+			return { hasError: true, message: action.message };
+		default:
+			return {};
+	}
+};
+
+const ReportPatternModal = ( { postId, onClose } ) => {
+	const [ state, dispatch ] = useReducer( reducer, {} );
+	const [ selectedOption, setOption ] = useState( '' );
+	const [ details, setDetails ] = useState( '' );
+
+	const { isLoading, reasons } = useSelect( ( select ) => {
+		const { getPatternFlagReasons, isLoadingPatternFlagReasons } = select( patternStore );
+
+		return {
+			isLoading: isLoadingPatternFlagReasons(),
+			reasons: getPatternFlagReasons(),
+		};
+	} );
+
+	const handleSubmit = ( event ) => {
+		event.preventDefault();
+
+		if ( ! selectedOption || ! details.length || state.isSubmitted || state.isSubmitting ) {
+			return;
+		}
+
+		dispatch( { status: 'submitting' } );
+
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/wporg-pattern-flag' ),
+			method: 'POST',
+			data: {
+				parent: postId,
+				'wporg-pattern-flag-reason': selectedOption,
+				excerpt: details,
+			},
+		} )
+			.then( () => {
+				dispatch( { status: 'submitted' } );
+			} )
+			.catch( ( err ) => {
+				dispatch( {
+					status: 'error',
+					message: err.message,
+				} );
+			} );
+	};
+
+	const handleClose = () => {
+		onClose( state.isSubmitted );
+	};
+
+	const mappedReasons = useMemo( () => {
+		if ( ! reasons ) {
+			return [];
+		}
+
+		return (
+			reasons
+				// We sort by id to give some control on the server side.
+				.sort( ( a, b ) => a.id - b.id )
+				.map( ( i ) => {
+					// We need to convert id to string to make the RadioControl match the selected item.
+					return { label: i.name, value: i.id.toString() };
+				} )
+		);
+	}, [ reasons ] );
+
+	const renderView = () => {
+		if ( isLoading ) {
+			return <Spinner />;
+		}
+
+		if ( state.isSubmitted ) {
+			return (
+				<p className="pattern-report-modal__copy">
+					{ __(
+						'Thank you for your report submission. We will review this pattern and act accordingly.',
```

Comment:
It feels like this wording needs some tweaking, but I'm not sure what exactly would be better. Maybe just make it simple, something like `Your report has been submitted`?

Reply:
Yeah fair enough. Let's go with `Your report has been submitted` and let the moderators/beta users help us tweak it.